### PR TITLE
[ALLUXIO-3139] Alluxio FS Shell cleanup

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3139

This is the first PR to fix a bunch in shell
- `AbstractCommand.getNumOfArgs()` as the required constant number of args is no more meaningful: we have multiple commands take multiple number of args (e.g., mount, help can take 0 or 1). Change `getNumOfArgs` to `checkArgs` so each command can check the args on demand 
- Add `getCommands()` to `AbstractShell`, so applications on Alluxio shell can list available commands.
- Totally remove `AbstractCommand` but move to `Command` interface
- some minor text cleanup

### CODE REVIEW

File: core/common/src/main/java/alluxio/cli/AbstractCommand.java [hunk 1]
```diff
@@ -36,19 +36,7 @@ protected AbstractCommand() {}
    * @param args the arguments for the command, excluding the command name and options
    * @throws InvalidArgumentException when arguments are not valid
```
Comment (gpang) [inline core/common/src/main/java/alluxio/cli/AbstractCommand.java]: add a return javadoc? What is the difference between returning `false` and throwing an exception?

File: shell/src/main/java/alluxio/cli/fs/command/PinCommand.java [hunk 2]
```diff
@@ -60,10 +60,7 @@ public String getDescription() {
   }
 
   @Override
-  public void validateArgs(String... args) throws InvalidArgumentException {
-    if (args.length < 1) {
-      throw new InvalidArgumentException(ExceptionMessage.INVALID_ARGS_NUM_INSUFFICIENT
-          .getMessage(getCommandName(), 1, args.length));
-    }
+  public void checkArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsNoLessThan(this, cl, 0);
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/fs/command/PinCommand.java]: fixed

File: shell/src/main/java/alluxio/cli/fs/command/RmCommand.java [hunk 3]
```diff
@@ -119,10 +115,7 @@ public String getDescription() {
   }
 
   @Override
-  public void validateArgs(String... args) throws InvalidArgumentException {
-    if (args.length < 1) {
-      throw new InvalidArgumentException(ExceptionMessage.INVALID_ARGS_NUM_INSUFFICIENT
-          .getMessage(getCommandName(), 1, args.length));
-    }
+  public void checkArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsNoLessThan(this, cl, 0);
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/fs/command/RmCommand.java]: fixed

**Comment by apc999** [comment]
@gpang  PTAL

**Pull request merged**